src/Tree.tsx

```diff
@@ -286,6 +286,7 @@ const generateNewTree = <
   let tempRecord: NodeRecord<TNodePublicState> | null = rootRecord;
 
   const useIdleCallback =
+    typeof window !== 'undefined' &&
     'requestIdleCallback' in window &&
```

Comment:
```suggestion
    typeof requestIdleCallback !== 'undefined'
```
I suppose, it should do the trick without an additional check. Could you update the PR?